File: 2574-left-and-right-sum-differences/2574-left-and-right-sum-differences.cpp
class Solution {
public:
    vector<int> leftRightDifference(vector<int>& nums) {
        int sumRight = 0;
        int sumLeft = 0;

        vector<int> res;

        for (auto n: nums) {
            sumRight += n;
        }

        for (int i = 0; i<nums.size(); i ++) {
            sumRight -= nums[i];
            res.push_back(abs(sumRight - sumLeft));
            sumLeft += nums[i];
        }

        return res;
    }
};

File: 2574-left-and-right-sum-differences/2574-left-and-right-sum-differences.py
class Solution:
    def leftRightDifference(self, nums: List[int]) -> List[int]:
        res = []
        right_sum = sum(nums)
        left_sum = 0
        for idx in range(len(nums)):
            tmp = nums[idx]
            right_sum -= tmp
            res.append(abs(right_sum-left_sum))
            left_sum += tmp
        return res